compute_sdk/globus_compute_sdk/sdk/diagnostic.py

```diff
@@ -0,0 +1,497 @@
+    of the base rich.print() as no_wrap defaults to None for stdout, inserting
+    line breaks when we don't want them (in tests for example)
+    """
+    Console(file=sys.stderr).print(f"[bold red]{s}[/bold red]", no_wrap=True)
+    print(s)
+
+
+def print_highlight(s: str):
+    get_console().print(f"[bold yellow]{s}[/bold yellow]", no_wrap=True)
+
+
+def os_info_cmd():
+    if sys.platform == "linux":
+        return cat("/etc/os-release")
+    elif sys.platform == "darwin":
+        return "sw_vers"
+    elif sys.platform == "win32":
+        return "systeminfo"
+    else:
+        return None
+
+
+def if_supported_on_os(cmd: str):
+    """
+    Misc commands that may exist for one or more platforms, but not for others.
+
+    A simpler 'supported or not' check that returns None if unavailable
+    """
+    if cmd:
+        BY_PLATFORM = {
+            "ip": ["linux"],
+            "lshw": ["linux"],
+            "df": ["linux", "darwin"],
+            "sysctl": ["linux", "darwin"],
+        }
+
+        cmd_base = cmd.split()[0]
+        if cmd_base in BY_PLATFORM:
+            return cmd if sys.platform in BY_PLATFORM.get(cmd_base, []) else None
+        else:
+            # If not specifically named here, assume it's runnable on all
+            return cmd
+    else:
+        return None
+
+
+def filter_by_os(cmd_list: list):
+    results = []
+    for cmd in cmd_list:
+        # if_supported_on_os only checks cmds in str form, not methods
+        if cmd and (not isinstance(cmd, str) or if_supported_on_os(cmd)):
+            results.append(cmd)
+
+    return results
+
+
+def print_endpoint_install_dir() -> str:
+    # Works for bare install or pyenv.  May not work for pipx
+    run_attempt = None
+    try:
+        run_attempt = subprocess.run("globus-compute-endpoint", capture_output=True)
+    except Exception:
+        # Treat errors as not found (FileNotFoundError raised on linux)
+        pass
+
+    if (
+        run_attempt is None
+        or bool(run_attempt.stderr)
+        and "command not found" in run_attempt.stderr.decode()
+    ):
+        return "globus-compute-endpoint is not installed"
+    else:
+        # `which` by itself may return a value even if gce is not installed.
+        # It will return the shim in pyenv, which contains info for other PY versions
+        ep_path = shutil.which("globus-compute-endpoint")
+        if ep_path:
+            return f"globus-compute-endpoint is installed at {ep_path}"
+        else:
+            return "globus-compute-endpoint is not installed"
+
+
+def connection_tests():
+    web_svc_url = get_web_service_url()
+    web_svc_host = urlparse(web_svc_url).netloc
+    amqp_svc_host = get_amqp_service_host()
+
+    return [
+        test_conn(web_svc_host, 443),
+        test_conn(amqp_svc_host, 5671),
+        test_ssl_conn(web_svc_host, 443),
+        test_ssl_conn(amqp_svc_host, 5671),
+        print_service_versions(web_svc_url),
+    ]
+
+
+def general_commands_list():
```

Comment:
Why not directly use a list?

Reply:
To be able to mock it for easier test.

Reply:
Easier mocking

Reply:
Your call, but you can directly mock a list.